com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationStager.java

```diff
@@ -133,11 +137,28 @@ public void bindDropletToApplication(DelegateExecution context, UUID appGuid) {
 
     public StepPhase stageApp(DelegateExecution context, CloudApplication app, StepLogger stepLogger) {
         UploadToken uploadToken = StepsUtil.getUploadToken(context);
-        if (uploadToken == null) {
-            return StepPhase.DONE;
+        if (uploadToken != null) {
+            stepLogger.debug(Messages.UPLOAD_TOKEN_FOR_APPLICATION_0_1, app.getName(), JsonUtil.toJson(uploadToken, true));
+            return stageApplication(context, app, stepLogger, uploadToken);
```

Comment:
Do `uploadToken.getPackageGuid()` here and remove the `stageApplication` method overload below.